components/basic-filters/src/test/java/org/datacleaner/beans/filter/StringPatternMatchFilterTest.java

```diff
@@ -28,37 +30,45 @@
 
 public class StringPatternMatchFilterTest extends TestCase {
 
-	public void testFilterSinglePattern() throws Exception {
-		StringPattern stringPattern = new RegexStringPattern("very simple email pattern", ".+@.+", true);
-		MockInputColumn<String> column = new MockInputColumn<String>("my col", String.class);
-		StringPatternFilter filter = new StringPatternFilter(column, new StringPattern[] { stringPattern },
-				MatchFilterCriteria.ANY);
+    private final DataCleanerConfiguration configuration = new DataCleanerConfigurationImpl();
 
-		assertEquals(ValidationCategory.VALID, filter.categorize(new MockInputRow().put(column, "kasper@eobjects.org")));
-		assertEquals(ValidationCategory.INVALID, filter.categorize(new MockInputRow().put(column, "kasper@")));
+    public void testFilterSinglePattern() throws Exception {
+        StringPattern stringPattern = new RegexStringPattern("very simple email pattern", ".+@.+", true);
+        MockInputColumn<String> column = new MockInputColumn<String>("my col", String.class);
+        StringPatternFilter filter = new StringPatternFilter(column, new StringPattern[] { stringPattern },
+                MatchFilterCriteria.ANY, configuration);
+        filter.init();
```

Comment:
And here the reason the missing annotations wasn't spotted.

Would it be viable to make a test that runs it through the engine to find these kinds of bugs?